third_party/gfxreconstruct/framework/decode/dive_block_data.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2025 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "decode/dive_block_data.h"
+#include "util/logging.h"
+
+#include <fstream>
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(decode)
+
+bool DiveBlockData::AddOriginalBlock(uint32_t index, uint64_t offset)
+{
+    if (original_blocks_map_locked_)
+    {
+        GFXRECON_LOG_ERROR("Cannot add more original blocks to a locked DiveBlockData object");
+        return false;
+    }
+
+    if (index != original_blocks_map_.size())
+    {
+        GFXRECON_LOG_ERROR(
+            "Unexpected block id mismatch with index: %d, expected index: %d", index, original_blocks_map_.size());
+        return false;
+    }
+
+    original_blocks_map_.push_back(BlockBytesLocation{ offset, 0 });
+
+    return true;
+}
+
+bool DiveBlockData::LockOriginalBlocksMap()
+{
+    if (original_blocks_map_locked_)
+    {
+        GFXRECON_LOG_INFO("Original block map already locked");
```

Comment:
i feel this log seems to be a bit redundant unless you want to give some warning here that something might goes wrong in this case?